packages/members-api/index.js

```diff
@@ -344,32 +345,44 @@ module.exports = function MembersApi({
 
             if (!member) {
                 throw new common.errors.BadRequestError({
-                    message: 'Cancel membership failed! Could not find member'
+                    message: 'Updating subscription failed! Could not find member'
                 });
             }
         } catch (err) {
             res.writeHead(401);
             return res.end('Unauthorized');
         }
-
         // Don't allow removing subscriptions that don't belong to the member
+        const plan = planName && stripe.findPlanByNickname(planName);
+        if (planName && !plan) {
+            throw new common.errors.BadRequestError({
+                message: 'Updating subscription failed! Could not find plan'
+            });
+        }
```

Comment:
Checks if plan name passed is valid and gets its data, throwing error if incorrect plan name is passed